pwa/src/templates/templateParts/mappingForm/MappingFormTemplate.tsx

```diff
@@ -80,19 +104,97 @@ export const MappingFormTemplate: React.FC<MappingFormTemplateProps> = ({ mappin
     basicFields.forEach((field) => setValue(field, mapping[field]));
 
     const newMapping = Object.entries(mapping.mapping).map(([key, value]) => ({ key, value }));
-    setMapping(newMapping);
+    setMappingKeyValue(newMapping);
+    setMappingCodeEditor(JSON.stringify(mapping.mapping.length ? mapping.mapping : {}, null, 2));
 
     const newUnset = Object.entries(mapping.unset).map(([key, value]) => ({ key, value }));
-    setUnset(newUnset);
+    setUnsetKeyValue(newUnset);
+    setUnsetCodeEditor(JSON.stringify(mapping.unset.length ? mapping.unset : {}, null, 2));
 
     const newCast = Object.entries(mapping.cast).map(([key, value]) => ({ key, value }));
-    setCast(newCast);
+    setCastKeyValue(newCast);
+    setCastCodeEditor(JSON.stringify(mapping.cast.length ? mapping.cast : {}, null, 2));
   };
 
   React.useEffect(() => {
     setIsLoading({ ...isLoading, mappingForm: createOrEditMapping.isLoading });
   }, [createOrEditMapping.isLoading]);
 
+  //Mappings
+  React.useEffect(() => {
+    if (mappingCodeEditor === "") return;
+
+    try {
+      JSON.parse(mappingCodeEditor);
+    } catch (e) {
+      return;
+    }
+
+    const jsonMapping = mappingCodeEditor && JSON.parse(mappingCodeEditor);
+
+    const newMapping = Object.entries(jsonMapping).map(([key, value]) => ({ key, value }));
+
+    if (!_.isEqual(mappingKeyValue, newMapping)) setMappingKeyValue(newMapping);
+  }, [mappingCodeEditor]);
+
+  React.useEffect(() => {
+    if (!watchMapping?.length) return;
+
+    const test = Object.assign({}, ...watchMapping.map(({ key, value }: any) => ({ [key]: value })));
+
+    setMappingCodeEditor(JSON.stringify(test, null, 2));
+  }, [watchMapping]);
+
+  //Unset
+  React.useEffect(() => {
+    if (unsetCodeEditor === "") return;
+
+    try {
+      JSON.parse(unsetCodeEditor);
+    } catch (e) {
+      return;
+    }
+
+    const jsonUnset = unsetCodeEditor && JSON.parse(unsetCodeEditor);
+
+    const newUnset = Object.entries(jsonUnset).map(([key, value]) => ({ key, value }));
+
+    if (!_.isEqual(unsetKeyValue, newUnset)) setUnsetKeyValue(newUnset);
+  }, [unsetCodeEditor]);
+
+  React.useEffect(() => {
+    if (!watchUnset?.length) return;
+
+    const test = Object.assign({}, ...watchUnset.map(({ key, value }: any) => ({ [key]: value })));
+
+    setUnsetCodeEditor(JSON.stringify(test, null, 2));
+  }, [watchUnset]);
+
+  //Cast
+  React.useEffect(() => {
+    if (castCodeEditor === "") return;
+
+    try {
+      JSON.parse(castCodeEditor);
+    } catch (e) {
+      return;
+    }
+
+    const jsonCast = castCodeEditor && JSON.parse(castCodeEditor);
+
+    const newCast = Object.entries(jsonCast).map(([key, value]) => ({ key, value }));
+
+    if (!_.isEqual(castKeyValue, newCast)) setCastKeyValue(newCast);
+  }, [castCodeEditor]);
```

Comment:
We're duplicating the same logic three times here. When you've implemented the other comment about simplifying the state, this might be easily simplified as well. If not, maybe we can try and pair-program a solution.